### PR TITLE
Pull in changes from Yook74's fork

### PR DESCRIPTION
**Pull request by misterhay**
Let's start looking at how to incorporate these back in.

### CODE REVIEW

**Comment by Yook74** [comment]
So one of the key design decisions I made was to separate this into a generic camera driver which could be used for any application and application code. My application code lives in [another repo ](https://github.com/International-Anglican-Church/visca-joystick) actually.

If we're going to publish this as a PyPI package, I'd recommend publishing just the generic driver, although I could see arguments for publishing it as an application too.

File: visca_over_ip/camera.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        self._sock.bind(('', port))
+        self._sock.settimeout(0.1)
+
+        self.num_missed_responses = 0
+        self.sequence_number = 0  # This number is encoded in each message and incremented after sending each message
+        self.num_retries = 5
+        self.reset_sequence_number()
+        self._send_command('00 01')  # clear the camera's interface socket
+
+    def _send_command(self, command_hex: str, query=False) -> Optional[bytes]:
```
Comment (Yook74) [inline visca_over_ip/camera.py:34]: This method is kind of... a lot. Sorry. It works pretty reliably though, and it really simplifies the code in all the public methods (the ones that don't start with "_")

**Comment by Yook74** [comment]
I haven't looked at your application code recently and I don't think much of it will be directly compatible with my `Camera` class. Hopefully it won't take much work to make it compatible though. Let me know how you'd like to divide labor on that effort (assuming that's the direction you'd like to go)

**Comment by misterhay** [comment]
I like your idea of separating the camera class, that's similar to what I've started with [camera.py](https://github.com/misterhay/VISCA-IP-Controller/blob/main/camera.py). I really like what you've done with blocking, retries, and response receipt attempts. It might be most efficient for us to add a few functions to the `Camera` class in your repo for submission to PyPI, and then convert this repo to just a GUI application that uses that class. Thoughts?

**Comment by Yook74** [comment]
I'd be happy to add methods to my Camera class. Let me know what you'd like to add. 

I'm fine with having the application and library code both on this repo in different directories or splitting them out into different repos. We will need some sort of home for the library code even if we publish it to PyPI though, so making this repo only application code without creating a new repo for the library code won't fly.

**Pull request merged**